packages/components/src/calendar-react-day-picker/styles.scss

```diff
@@ -0,0 +1,448 @@
+      border-end-end-radius: 0;
+    }
+  }
+
+  // Middle of date range
+  &__range-middle &__day-button {
+    &::before {
+      background-color: var(--a8c-components-calendar-range-middle-background-color);
+      border-radius: 0;
+
+      // Render a top and bottom transparent border to highlight the selected
+      // day in forced-colors (high-contrast) mode, since the background is not
+      // visible.
+      border-width: 1px 0;
+      border-color: transparent;
+      border-style: solid;
+    }
+  }
+
+  // Range end button, but not when start and end are the same day.
+  &__range-end:not(&__range-start) &__day-button {
+    // Apply border-radius changes to the button itself too, so that the focus
+    // ring follows the same shape as the button's background.
+    &, &::before {
+      border-start-start-radius: 0;
+      border-end-start-radius: 0;
+    }
+  }
+
+  /*
+   * RANGE PREVIEW (range calendar only)
+   *
+   * The preview is rendered in the button's ::after pseudo-element, so that it
+   * can be rendered over the button's contents.
+   * The selection preview is shown with a dashed border. In order to have
+   * control over the dash pattern (especially the seams between days), the
+   * dashed borders are rendered as SVGs via the url() CSS function.
+   * Since SVGs rendered in the url() function don't seem to be able to access
+   * CSS variables, we're using the SVGs as masks, and using `background-color`
+   * to consume the accent color variable.
+   */
+
+  &__day--preview:not(&__range-start):not(&__range-end) &__day-button::after {
+    background-color: var(--a8c-components-calendar-preview-border-color);
+
+    // Since the background is not visible in forced-colors (high-contrast)
+    // mode, add a transparent border which is still going to be masked, and
+    // thus visible in forced-colors mode.
+    border: 1px solid transparent;
+  }
+  &__day--preview:is(&__range-middle):not(&__range-start):not(&__range-end) &__day-button::before {
+    // Remove the transparent border shown on the middle of the range
+    // in forced-colors (high-contrast) mode, to allow for the dashed border
+    // to be visible.
+    border: none;
+  }
+
+&__day--preview:not(&__day--preview-start):not(&__day--preview-end):not(&__range-start):not(&__range-end) &__day-button::after {
+    // SVG with dashed top/bottom lines. The dash pattern is chosen to make sure
+    // that multiple days in a row show a seamless dashed border.
+    // The stylelint rule is disabled because we need to wrap the SVG in quotes
+    // to disambiguate from the Sass url() function.
+    /* stylelint-disable-next-line function-url-quotes */
+    mask-image: url("data:image/svg+xml, %3Csvg viewBox='0 0 32 32' xmlns='http://www.w3.org/2000/svg' fill='white' stroke='black' stroke-dasharray='3.9 4' stroke-dashoffset='2' stroke-width='1'%3E%3Cline x1='0' y1='0.5' x2='100' y2='0.5' /%3E%3Cline x1='0' y1='31.5' x2='100' y2='31.5' /%3E%3C/svg%3E");
```

Comment:
I don't think this would be very compatible with a [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/CSP) if that's something we want to consider.